src/Generalize.js

```diff
@@ -264,6 +256,19 @@ L.MarkerGeneralizeGroup = L.FeatureGroup.extend({
             levelsCallback();
         }
 
+        function getMarkerLevel(marker, index) {
+            var markerType = marker.options.type;
+            if (!markerType) {
+                return levels[index];
+            }
+            for (var i = 0; i <= levels.length; i++) {
```

Comment:
Start iterating from index value rather than 0 and remove unnecessary condition on the next line.